### PR TITLE
ENG-277: create packs page

### PR DESCRIPTION
**Pull request by stormcloud266**
Adds packs collection page. Updates puzzle components to be used with packs. Add packs link to header.

### CODE REVIEW

**Review by chair28980 — APPROVED** [review]
The UI lgtm! 🍏

File: lib/utils.ts [hunk 1]
```diff
@@ -29,3 +38,28 @@ export const toHex = (num: number) => {
   const val = Number(num);
   return "0x" + val.toString(16);
 };
+
+// Thumbnail grid
+export const isTypePack = (
+  data: ThumbnailPack | ThumbnailPuzzle
+): data is ThumbnailPack => {
+  return (data as ThumbnailPack).pack_name !== undefined;
+};
```
Comment (illepic) [inline lib/utils.ts:48]: Use [Zod](https://zod.dev/), it's already installed and [used to typeguard a Guess.](https://github.com/infinity-keys/ik-landing/blob/main/lib/types.ts#L5-L15) like you're doing here manually for `ThumbnailPack`.

File: components/puzzles/layout.tsx [hunk 2]
```diff
@@ -11,9 +11,16 @@ import PuzzleThumbnail from "@components/puzzle-thumbnail";
...
 import { PAGINATION_COUNTS } from "@lib/constants";
-
 import { PuzzleLayoutType } from "@lib/types";
-import { PageProps } from "pages/puzzles/[[...puzzlesArgs]]";
+import { collectionBaseUrl, isTypePack, thumbnailData } from "@lib/utils";
+
+import { GetAllPacksQuery, PublicPuzzlesQuery } from "@lib/generated/graphql";
+
+export interface PageProps {
+  puzzles: PublicPuzzlesQuery["puzzles"] | GetAllPacksQuery["packs"];
```
Comment (illepic) [inline components/puzzles/layout.tsx]: NICE

File: lib/types.ts [hunk 3]
```diff
@@ -33,3 +34,6 @@ export enum PuzzleLayoutType {
   List = "list",
   Unknown = "unknown",
 }
+
+export type ThumbnailPuzzle = PublicPuzzlesQuery["puzzles"][0];
+export type ThumbnailPack = GetAllPacksQuery["packs"][0];
```
Comment (illepic) [inline lib/types.ts:40]: NICE

**Review by illepic — CHANGES_REQUESTED** [review]
The only thing here is to try zod.

**Review by illepic — APPROVED** [review]

**Pull request merged**